src/policy/fees.cpp

```diff
@@ -504,10 +507,27 @@ CBlockPolicyEstimator::CBlockPolicyEstimator()
     feeStats = std::unique_ptr<TxConfirmStats>(new TxConfirmStats(buckets, bucketMap, MED_BLOCK_PERIODS, MED_DECAY, MED_SCALE));
     shortStats = std::unique_ptr<TxConfirmStats>(new TxConfirmStats(buckets, bucketMap, SHORT_BLOCK_PERIODS, SHORT_DECAY, SHORT_SCALE));
     longStats = std::unique_ptr<TxConfirmStats>(new TxConfirmStats(buckets, bucketMap, LONG_BLOCK_PERIODS, LONG_DECAY, LONG_SCALE));
+
+    // If the fee estimation file is present, read recorded estimations
+    fs::path est_filepath = GetDataDir() / FEE_ESTIMATES_FILENAME;
+    if (est_filepath.empty()) return;
+
+    CAutoFile est_file(fsbridge::fopen(est_filepath, "rb"), SER_DISK, CLIENT_VERSION);
+    if (est_file.IsNull() || !Read(est_file)) {
+        LogPrintf("%s: Failed to read fee estimates from %s\n", __func__, est_filepath.string());
+    }
 }
 
 CBlockPolicyEstimator::~CBlockPolicyEstimator()
 {
+    // If the fee estimation file is present, record current estimations
+    fs::path est_filepath = GetDataDir() / FEE_ESTIMATES_FILENAME;
+    if (est_filepath.empty()) return;
+
+    CAutoFile est_file(fsbridge::fopen(est_filepath, "wb"), SER_DISK, CLIENT_VERSION);
+    if (est_file.IsNull() || !Write(est_file)) {
```

Comment:
Why are you moving this to the destructor? Are we sure that nothing happens to the fee estimator's state between  https://github.com/bitcoin/bitcoin/pull/18766/files#diff-c865a8939105e6350a50af02766291b7R236 and https://github.com/bitcoin/bitcoin/pull/18766/files#diff-c865a8939105e6350a50af02766291b7R294?

Reply:
Right. What do you think of instead exposing a `RecordEstimation()` method to the `CBlockPolicyEstimator` ?

Reply:
Was there a reason you didn't like the flush() method in https://github.com/jnewbery/bitcoin/tree/pr18766.1?

Reply:
Ok didn't see this. Had a version with both the FlushUnconfirmed() and RecordEstimation() separated locally. Grouping them and pushing in a minute !